src/axom/quest/MeshViewUtil.hpp

```diff
@@ -0,0 +1,778 @@
+class MeshViewUtil
+{
+public:
+  using CoordsViewsType =
+    axom::StackArray<axom::ArrayView<double, DIM, MemSpace>, DIM>;
+  using ConstCoordsViewsType =
+    axom::StackArray<axom::ArrayView<const double, DIM, MemSpace>, DIM>;
+
+  //@{
+  //@name Setting up
+  /*!
+    @brief Constructor
+    @param [in] bpDomain Blueprint single domain.
+    @param [in] topologyName Name of topology in the domain.
+
+    If \a topologyName is omitted, use the first topology,
+
+    The topology dimension must match DIM.
+  */
+  MeshViewUtil(conduit::Node& bpDomain, const std::string& topologyName = "")
+    : m_dom(&bpDomain)
+    , m_topologyName(topologyName)
+  {
+    if(m_topologyName.empty())
+    {
+      m_topologyName = bpDomain.fetch_existing("topologies")[0].name();
+    }
+
+    const std::string topoPath = "topologies/" + m_topologyName;
+    m_topology = &m_dom->fetch_existing(topoPath);
+
+    const std::string coordsetPath =
+      "coordsets/" + m_dom->fetch_existing(topoPath + "/coordset").as_string();
+    m_coordset = &m_dom->fetch_existing(coordsetPath);
+
+    m_cdom = m_dom;
+    m_ctopology = m_topology;
+    m_ccoordset = m_coordset;
+
+    if(!isValid())
+    {
+      SLIC_ERROR(
+        "Invalid domain in MeshViewUtil.  Domain must be blueprint-valid,"
+        " structured, with exlicit coordinates and dimension equal to "
+        " the template DIM parameter.");
+    }
+
+    computeCoordsDataLayout();
+  }
+
+  /*!
+    @brief Constructor
+    @param [in] bpDomain const Blueprint single domain.
+    @param [in] topologyName Name op topology in the domain.
+  */
+  MeshViewUtil(const conduit::Node& bpDomain, const std::string& topologyName)
+    : m_cdom(&bpDomain)
+    , m_topologyName(topologyName)
+  {
+    const std::string topoPath = "topologies/" + m_topologyName;
+    SLIC_ASSERT(m_cdom->has_path(topoPath));
+    m_ctopology = &m_cdom->fetch_existing(topoPath);
+
+    const std::string coordsetPath =
+      "coordsets/" + m_cdom->fetch_existing(topoPath + "/coordset").as_string();
+    m_ccoordset = &m_cdom->fetch_existing(coordsetPath);
+
+    if(!isValid())
+    {
+      SLIC_ERROR(
+        "Invalid domain in MeshViewUtil.  Domain must be blueprint-valid,"
+        " structured, with exlicit coordinates. and dimension equal to "
+        " the template DIM parameter.");
+    }
+
+    computeCoordsDataLayout();
+  }
+
+  /*!
+    @brief Whether the domain is a valid one for this utility.
+
+    In addition to checking for blueprint validity (unless \a
+    checkBlueprint is false), this method checks against its own
+    requirements.
+  */
+  bool isValid(bool checkBlueprint = false) const
+  {
+    bool rval = true;
+    if(checkBlueprint)
+    {
+      conduit::Node info;
+  #ifdef AXOM_USE_MPI
+      rval = rval &&
+        conduit::blueprint::mpi::verify("mesh", *m_cdom, info, MPI_COMM_WORLD);
+  #else
+      rval = rval && conduit::blueprint::verify("mesh", *m_cdom, info);
+  #endif
```

Comment:
It might be helpful to optionally print the `info` node in case of problems (or dump it to disk)

Reply:
Implementing.